shared/actions/fs/platform-specific.desktop.js

```diff
@@ -287,10 +287,32 @@ function installCachedDokan() {
   })
 }
 
+function uninstallDokan(uninstallString: string): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    logger.info('Invoking dokan uninstaller')
+    try {
+      execSync(uninstallString, {windowsHide: true})
```

Comment:
we shouldn't ever make sync calls, the entire ui will just hang if we do this

Reply:
Yeah that's what we were talking about in chat.

Reply:
i think this could be something like
```
const uninstallDokanSaga = (state: TypedState) => {
    const uninstallString = Constants.kbfsUninstallString(state)
    if (!uninstallString) {
        return 
    }
    logger.info('Invoking dokan uninstaller')
    return new Promise(resolve => {
        try {
            exec(uninstallString, {windowsHide: true}, resolve)
        }  catch (e) {
            logger.error('uninstallDokan caught', err)
            resolve()
        }
    }).then(() => FsGen.createFuseStatus())
}

...
    yield Saga.actionToPromise(FsGen.uninstallKBFSConfirm, uninstallDokanSaga)

```

Reply:
huh, I thought that was the whole point of a Promise, to synchronize lengthy things.

Anyway, thanks Nojima, passing resolve as the callback is what I was missing, I guess my promise wasn't ever resolving when i tried it that way.

Reply:
this is incorrect. Promises are just a way to control async stuff.  Please read https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises